Packs/TrendMicroVisionOne/Integrations/TrendMicroVisionOne/TrendMicroVisionOne.py

```diff
@@ -0,0 +1,1053 @@
+        :param base_url: Base server address with suffix, for example: https://example.com/api/v2/.
+        :type api_key: ``str``
+        :param api_key: api token to access the api data.
+        :type verify: ``bool``
+        :param verify: Whether the request should verify the SSL certificate.
+        :return: returns None
+        :rtype: ``None``
+        """
+        super().__init__(base_url=base_url)
+        self.base_url = base_url
+        self.api_key = api_key
+        self.status = None
+
+    def http_request(self, method: str, url_suffix: str, json_data=None, params=None, data=None) -> Any:
+        """
+        Override http_request method from BaseClient class. This method will print an error based on status code
+        and exceptions.
+        :type method: ``str``
+        :param method: The HTTP method, for example: GET, POST, and so on.
+        :type url_suffix: ``str``
+        :param url_suffix: The API endpoint.
+        :type json_data: ``dict``
+        :param json_data: The dictionary to send in a 'POST' request.
+        :type params: ``dict``
+        :param params: URL parameters to specify the query.
+        :type data: ``dict``
+        :param data: The data to send in a 'POST' request.
+        :return: response data
+        :rtype: ``dict`` or ``str`` or ``requests.Response``
+        """
+        header = {
+            "Authorization": "Bearer {token}".format(token=self.api_key),
+            "Content-Type": f'{CONTENT_TYPE_JSON}{";charset=utf-8"}'
+        }
+        try:
+            response = self._http_request(method=method, full_url=f'{self.base_url}{url_suffix}', retries=3, json_data=json_data,
+                                          params=params, headers=header, resp_type='response',
+                                          ok_codes=(200, 201), data=data)
+        except DemistoException as error:
+            demisto.error(error.message)
+            return_error(error.message)
+        if response.ok:
+            demisto.info(SUCCESS_RESPONSE.format(url=f'{self.base_url}{url_suffix}', status=response.status_code))
+            self.status = response.status_code
+            content_type = response.headers.get('Content-Type', '')
+            if content_type.__contains__(CONTENT_TYPE_JSON):
+                # Handle empty response
+                if response.text == EMPTY_STRING:
+                    return response
+                else:
+                    return response.json()
+            else:
+                return response
+
+    def status_check(self, data: Dict[str, Any]) -> Any:
+        """
+        Check the status of particular task.
+        :type data: ``dict``
+        :param method: Response data to received from the end point.
+        :return: task status response data.
+        :rtype: ``Any``
+        """
+        task_id = data.get("actionId", {})
+        params = {"actionId": task_id}
+        flag = True
+        while flag:
```

Comment:
@mikedgibson This is not how you're supposed to use `ScheduledCommands`.
The idea behind `ScheduledCommand` is that you return a result that schedules another run of the command in the future (e.g. in 30 seconds). A loop makes no sense, as the `ScheduledCommand` architecture provides 
 the logical looping you require.
In this case, the intent is to call the command again with the `task_id` and to check whether it finished or the task or not. If it didn't finish the task, it should return a `ScheduledCommand` saying it didn't finish, and that it'll try again in 30 seconds. If it did finish the task, then it shouldn't return a `ScheduledCommand` as there's no need for further polling.

I invite you to take a look at the `AutoFocus V2` integration to see how it's generically handled there. You can copy a lot of the code from there - just keep in mind your command arguments should support the `PollingArgs` you are passing.

If you need further assistance in using `PollingCommand` we can do a short session together.

Reply:
Our development firm updated the use of ScheduledCommand. If this is still not the correct use then we would appreciate the opportunity to take you up on your offer for a short session. Thanks!!